src/app/view/components/dot-add-persona-dialog/dot-create-persona-form/dot-create-persona-form.component.ts

```diff
@@ -0,0 +1,117 @@
+import { Component, EventEmitter, OnDestroy, OnInit, Output } from '@angular/core';
+import { Site, SiteService } from 'dotcms-js';
+import { take, takeUntil } from 'rxjs/operators';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { DotMessageService } from '@services/dot-messages-service';
+import * as _ from 'lodash';
+import { Subject } from 'rxjs';
+import { DotFileUpload } from '@models/dot-file-upload/dot-file-upload.model';
+import { DotCMSTempFile } from 'dotcms-models';
+
+@Component({
+    selector: 'dot-create-persona-form',
+    templateUrl: './dot-create-persona-form.component.html',
+    styleUrls: ['./dot-create-persona-form.component.scss']
+})
+export class DotCreatePersonaFormComponent implements OnInit, OnDestroy {
+    @Output() value: EventEmitter<FormGroup> = new EventEmitter();
+
+    form: FormGroup;
+    imageName: string;
+    messagesKey: { [key: string]: string } = {};
+
+    private destroy$: Subject<boolean> = new Subject<boolean>();
+
+    constructor(
+        public dotMessageService: DotMessageService,
+        private fb: FormBuilder,
+        private siteService: SiteService
+    ) {}
+
+    ngOnInit() {
+        this.dotMessageService
+            .getMessages([
+                'modes.persona.upload.file',
+                'modes.persona.name',
+                'modes.persona.key.tag',
+                'dot.common.choose',
+                'dot.common.remove',
+                'modes.persona.host',
+                'modes.persona.name.error.required'
+            ])
+            .pipe(take(1))
+            .subscribe((messages: { [key: string]: string }) => {
+                this.messagesKey = messages;
+            });
+        this.initPersonaForm();
+    }
+
+    ngOnDestroy(): void {
+        this.destroy$.next(true);
+        this.destroy$.complete();
+    }
+
+    /**
+     * Handle the response of the p-fileUpload to update the form.
+     *
+     * @param {DotFileUpload} event
+     * @memberof DotCreatePersonaFormComponent
+     */
+    onFileUpload(event: DotFileUpload) {
+        const response = JSON.parse(event.xhr.response);
+        const tempFile: DotCMSTempFile = response.tempFiles[0];
```

Comment:
maybe we can do all this in a util class, something like:

```suggestion
     getTemoFile(dotFileUpload: DotFileUpload): DotCMSTempFile {
                const response = JSON.parse(dotFileUpload.xhr.response);
               return response.tempFiles[0];
      }
```